### ISSUE TITLE
#pragma GCC diagnostic not allowed inside functions

### ISSUE THREAD

**Issue by vext01** [SWI-Prolog/swipl-devel#250]
On OpenBSD-current with SWI-Prolog-7.4.2 and GCC-4.9.4:

```
...
cc -c -O2 -pipe  -ftrampolines   -I/usr/local/include -O2 -pipe -ftrampolines  -pthread -fPIC -I/usr/local/include -I/usr/local/pobj/sw
i-prolog-7.4.2/swipl-7.4.2/src/../include -I. -I/usr/local/pobj/swi-prolog-7.4.2/swipl-7.4.2/packages/xpce/src -DHAVE_CONFIG_H -D__SWI_
PROLOG__ ker/goodies.c -o ker/goodies.o                                                                                                
ker/goodies.c: In function 'scanstr':                                                                                                  
ker/goodies.c:1001: error: #pragma GCC diagnostic not allowed inside functions                                                         
ker/goodies.c:1002: error: #pragma GCC diagnostic not allowed inside functions                                                         
ker/goodies.c:1099: error: #pragma GCC diagnostic not allowed inside functions                                                         
gmake[3]: *** [Makefile:131: ker/goodies.o] Error 1                                                                                    
gmake[3]: Leaving directory '/usr/local/pobj/swi-prolog-7.4.2/swipl-7.4.2/packages/xpce/src'                                           
gmake[2]: *** [Makefile:11: all] Error 2                                                                                               
gmake[2]: Leaving directory '/usr/local/pobj/swi-prolog-7.4.2/swipl-7.4.2/packages/xpce'                                               
gmake[2]: Entering directory '/usr/local/pobj/swi-prolog-7.4.2/swipl-7.4.2/packages/sgml'                                              
cc -O2 -pipe  -ftrampolines   -I/usr/local/include -O2 -pipe -ftrampolines  -pthread -fPIC -I/usr/local/include -I/usr/local/pobj/swi-p
rolog-7.4.2/swipl-7.4.2/src/../include  -DHAVE_CONFIG_H -D__SWI_PROLOG__ -I.   -c -o parser.o parser.c                                 
...
```

Cheers

**Comment by vext01** [comment]
Notice how the build continues afterwards, as well as in #249. It would be better if the build stopped.

**Comment by JanWielemaker** [comment]
Is fixed in 7.5.x series.  Will be fixed with 7.6, which will follow 7.4.2 due to unavoidable incompatibilities.

**Comment by JanWielemaker** [comment]
It is a deliberate choice not to stop the build if a package fails.  In the end the build reports which packages 
have failed building.  For many people it is ok if some package fails building and some are notoriously hard on some systems (e.g., JPL).
